.github/workflows/ci.yml

```diff
@@ -708,3 +708,75 @@ jobs:
           df -h
           ps auxf
           cat /tmp/surrealdb.log || true
+
+  db-upgrade-v1-0-1:
```

Comment:
I would suggest a matrix job so we can reuse the logic.

you can pass the specific make command to the step like this:

```

  db-upgrade:
    name: Database Upgrade from ${{ matrix.version }}
    runs-on: ubuntu-latest
    strategy:
      fail-fast: false
      matrix:
        include:
          - version: 1.0.1
            command: ci-database-upgrade-1-0-1
          - version: 1.1.1
            command: ci-database-upgrade-1-1-1
          - version: 1.2.1
            command: ci-database-upgrade-1-2-1
    steps:

      - name: Install stable toolchain
        uses: dtolnay/rust-toolchain@stable
        with:
          toolchain: 1.75.0

      - name: Checkout sources
        uses: actions/checkout@v4

      - name: Setup cache
        uses: Swatinem/rust-cache@v2
        with:
          save-if: ${{ github.ref == 'refs/heads/main' }}

      - name: Install cargo-make
        run: cargo install --debug --locked cargo-make

      - name: Test DB upgrade
        run: cargo make ${{ matrix.command }}
```

Reply:
Absolutely, I didn't feel comfortable copying/pasting these lines...
Thanks for the example, I am doing that!

Reply:
I went further. There is no need for three separate CI-tasks, they can even run on the same build and task. Rather than 3 CI-tasks we now just have 3 tests

Reply:
That's even better. This enables us to run all locally with a single command 👌